plugins/user-settings/src/components/useUserProfileInfo.ts

```diff
@@ -22,16 +22,34 @@ import {
 } from '@backstage/core-plugin-api';
 import { useEffect } from 'react';
 import useAsync from 'react-use/esm/useAsync';
+import { catalogApiRef } from '@backstage/plugin-catalog-react';
 
 /** @public */
 export const useUserProfile = () => {
   const identityApi = useApi(identityApiRef);
   const alertApi = useApi(alertApiRef);
+  const catalogApi = useApi(catalogApiRef);
 
   const { value, loading, error } = useAsync(async () => {
+    const identityProfile = await identityApi.getProfileInfo();
+    const backStageIdentity = await identityApi.getBackstageIdentity();
+    const catalogProfile = await catalogApi.getEntityByRef(
+      backStageIdentity.userEntityRef,
+    );
+    if (
+      identityProfile.picture === undefined &&
+      catalogProfile?.metadata?.annotations?.[
+        'backstage.io/user-profile-picture'
+      ]
```

Comment:
```suggestion
     isUserEntity(catalogProfile) && catalogProfile.spec.profile?.picture
```
I have meant the picture in the entity spec.
https://backstage.io/docs/features/software-catalog/descriptor-format#kind-user

Reply:
oops ;)

Reply:
fixed